Buildings/Fluid/CHPs/OrganicRankine/BaseClasses/FixedEvaporating.mo

```diff
@@ -0,0 +1,204 @@
+    "Hot fluid temperature at pinch point";
+  Modelica.Units.SI.TemperatureDifference dTPinEva(start = dTPinEva_set)
+    "Pinch point temperature difference of evaporator";
+  Modelica.Units.SI.ThermodynamicTemperature TWorCon
+    "Working fluid condensing temperature";
+  Modelica.Units.SI.ThermodynamicTemperature TColOut
+    "Fluid temperature out of the condenser";
+  Modelica.Units.SI.ThermodynamicTemperature TColPin(
+    start = 300)
+    "Cold fluid temperature at pinch point";
+  Modelica.Units.SI.MassFlowRate mWor_flow
+    "Mass flow rate of the working fluid"
+    annotation (Dialog(group="Cycle"));
+  Modelica.Blocks.Logical.Hysteresis hys(
+    uLow = mWor_flow_min,
+    uHigh = mWor_flow_min + mWor_flow_hysteresis,
+    u = mWor_flow_internal)
+    "Hysteresis for turning off cycle when working fluid flow too low";
+
+protected
+  Modelica.Units.SI.MassFlowRate mWor_flow_internal(
+    start = (mWor_flow_max + mWor_flow_min) / 2)
+    "Working fluid flow rate, intermediate variable";
+  Modelica.Units.SI.ThermodynamicTemperature THotPin_internal
+    "Hot fluid temperature at pinch point, intermedaite variable";
+  Modelica.Units.SI.ThermodynamicTemperature THotOut_internal
+    "Hot fluid outgoing temperature, intermediate variable";
+  Modelica.Units.SI.HeatFlowRate QEva_flow_internal
+    "Evaporator heat flow rate, intermediate variable";
+
+equation
+
+  assert(not (TWorCon > TWorEva - 1 and ena),
+"*** In " + getInstanceName() +
+": Working fluid condensing temperature is too high and close to evaporating temperature.
+This is likely caused by the flow rate of cold fluid in the condenser being too low
+when the ORC is on.");
+
+  assert(not (pCon < 101325 - 1 and ena and useLowCondenserPressureWarning),
+"*** In " + getInstanceName() +
+": Working fluid condensing pressure is lower than 101325 Pa.
+If this is intended, set useLowCondenserPressureWarning = false to turn off this warning.",
+level=AssertionLevel.warning);
+
+  // Evaporator
+  QEva_flow = mHot_flow * cpHot * (THotOut - THotIn);
+  QEva_flow = mWor_flow * (hPumOut - hExpInl);
+  // Pinch point
+  (THotPin - THotOut) * (hExpInl - hPumOut)
+  = (hPinEva - hPumOut) * (THotIn - THotOut);
+  dTPinEva = THotPin - TWorEva;
+
+  // Evaporator internal computation
+  QEva_flow_internal = mHot_flow * cpHot * (THotOut_internal - THotIn);
+  QEva_flow_internal = mWor_flow_internal * (hPumOut - hExpInl);
+  (THotPin_internal - THotOut_internal) * (hExpInl - hPumOut)
+  = (hPinEva - hPumOut) * (THotIn - THotOut_internal);
+  dTPinEva_set = THotPin_internal - TWorEva;
+
+  // Condenser
+  QCon_flow = mCol_flow * cpCol * (TColOut - TColIn);
+  QCon_flow = mWor_flow * (hExpOut - hPumInl);
+  // Pinch point
+  (TColPin - TColIn) * (hExpOut - hPumInl)
+  = (hPinCon - hPumInl) * (TColOut - TColIn);
+  dTPinCon = TWorCon - TColPin;
+
+  // Other components
+  PExp = mWor_flow * (hExpOut - hExpInl);
+  PPum = mWor_flow * (hPumOut - hPumInl);
+  mWor_flow =
+    if ena and hys.y
```

Comment:
`ena and hys.y` is equal to `on_actual`, so you should use `on_actual` here.

Reply:
Fixed in be47fe9.